### ISSUE TITLE
How to manage post-build-hook phase with the current stable Cabal version?

### ISSUE THREAD

**Issue by danfran** [haskell/cabal#3000]
Hi,

based on #2978 (that looks fixed now on the master version of Cabal), I am still looking for a solution with the current stable version 1.22.4.0 (and older versions) to manage a post-build-hook inside a sandbox.

One assumption first:
The sandbox-hash looks built in `Distribution.Client.Sandbox` via `sandboxBuildDir` (Jenkins hash algorithm). AFAIK it uses the current Cabal instance (/path/to/.cabal) as input, so every single project that uses the same Cabal instance will end up for being built in the same (relative) directory name. 

Example: `dist/dist-sandbox-8121acba` 

If the assumption is correct:

1) Do I have to implement the same algortihm in every post-build-hook to "infer" the current sandbox-path?

2) How a post-build-hook can be aware that the building is running in a  sandbox?

Thanks
Daniele

**Comment by ezyang** [comment]
Have you tried printing out LocalBuildInfo and seeing if anything in there looks reasonable? Since you are trying to workaround the issue, it seems OK to copypaste any relevant code from cabal-install to get it working.

**Comment by danfran** [comment]
I printed out the LocalBuildInfo that it turned out to be really huge. So I have extracted only the info relative to a module (Arithmetic):

```
{exeName = "Arithmetic", modulePath = "Arithmetic.hs", buildInfo = BuildInfo 
{buildable = False, buildTools = [], cppOptions = [], ccOptions = [],
 ldOptions = [], pkgconfigDepends = [], frameworks = [], 
cSources = [], jsSources = [], hsSourceDirs = ["src"], otherModules = [], 
defaultLanguage = Nothing, otherLanguages = [], defaultExtensions = [], 
otherExtensions = [], oldExtensions = [], extraLibs = [], extraGHCiLibs = [], 
extraLibDirs = [], includeDirs = [], includes = [], installIncludes = [], options = [],
 profOptions = [], sharedOptions = [], customFieldsBI = [], targetBuildDepends = [], 
targetBuildRenaming = fromList []}}
```

These are other values I get from `Simple.postBuild.Args` => []
and:

```
BuildFlags {buildProgramPaths = [], buildProgramArgs = [], buildDistPref = Flag "dist", 
buildVerbosity = Flag Normal, buildNumJobs = Flag (Just 8), buildArgs = []}
```

As you can see there is pretty much nothing significative in there. BTW I used the current git-master version of Cabal.

So to workaround the problem, at the moment I am using something like this:

```
myBuildHook :: Args -> BuildFlags -> PackageDescription -> LocalBuildInfo -> IO ()
myBuildHook apps _ _ pkg lbi =
  do builtApps <- getBuiltApps lbi
     forM_ builtApps $ — do stuff here   
  where getBuiltApps = filterM (doesDirectoryExist . someTransformationHereToGetThePathFromLBI)
```

**Comment by danfran** [comment]
I have just upgraded to Cabal 1.22.6.0 and the issues above look gone.

Does the version 1.22.6.0 include the changes suggested here: #2978 ?

I cannot find a complete changelog.

Thanks

**Comment by ezyang** [comment]
I don't know, someone would have to dig through the history and see what might have fixed it. I searched for references to sandbox back to the beginning of 2014 but didn't see anything. Anyway, since it is fixed in Cabal 1.22.6 I'm going to close this.
